src/openforms/js/components/admin/form_design/variables/ServiceFetchConfigurationForm.js

```diff
@@ -0,0 +1,332 @@
+                    {...formik.getFieldProps('body')}
+                  />
+                </Field>
+              </FormRow>
+            ) : null}
+          </Fieldset>
+
+          <Fieldset
+            title={
+              <FormattedMessage
+                defaultMessage="Data extraction"
+                description="Service fetch configuration modal data extraction fieldset title"
+              />
+            }
+            extraClassName="admin-fieldset"
+          >
+            <FormRow>
+              <Field
+                name={'dataMappingType'}
+                required
+                label={
+                  <FormattedMessage
+                    defaultMessage="Mapping expression language"
+                    description="Service fetch configuration modal form mapping expression language field label"
+                  />
+                }
+              >
+                <Select
+                  name="dataMappingType"
+                  choices={EXPRESSION_MAPPING_LANGUAGES}
+                  value={formik.values.dataMappingType}
+                  allowBlank
+                  {...formik.getFieldProps('dataMappingType')}
+                />
+              </Field>
+            </FormRow>
+
+            <FormRow>
+              <Field
+                name={'mappingExpression'}
+                required
+                label={
+                  <FormattedMessage
+                    defaultMessage="Mapping expression"
+                    description="Service fetch configuration modal form mapping expression field label"
+                  />
+                }
+              >
+                {formik.values.dataMappingType === 'JsonLogic' ? (
+                  <JsonWidget
+                    name="mappingExpression"
+                    logic={formik.values.mappingExpression}
+                    cols={20}
+                    {...formik.getFieldProps('mappingExpression')}
+                  />
+                ) : (
+                  <TextInput
+                    // Explicitly cast object to strings, in case the JsonWidget was used before
+                    value={
+                      typeof formik.values.mappingExpression === 'object'
+                        ? JSON.stringify(formik.values.mappingExpression)
+                        : formik.values.mappingExpression
+                    }
+                    {...formik.getFieldProps('mappingExpression')}
+                  />
+                )}
+              </Field>
+            </FormRow>
+          </Fieldset>
+
+          <SubmitRow>
+            {selectExisting ? (
+              <>
+                <ActionButton
+                  name="_save_as_new"
+                  text={intl.formatMessage({
+                    description: 'Save as new service fetch configuration button label',
+                    defaultMessage: 'Save as new',
+                  })}
+                  onClick={formik.handleSubmit}
+                />
```

Comment:
I'd expect a prop `type="submit"` and then in the parent `<form>` tag something like `<form onSubmit={formik.handleSubmit}>`, so that submitting the form with the <kbd>Enter</kbd> key also works.